src/Components/TableComponents/NewMatchesTableRow.js

```diff
@@ -1,27 +1,44 @@
-import React from 'react';
-import { Td } from '@patternfly/react-table';
-import PropTypes from 'prop-types';
-import { useCallback, useState } from 'react';
-import { useVerifyInput } from '../useVerifyInput';
+import React, { useCallback, useState } from 'react';
 import { useMutation } from '@apollo/client';
-import { EDIT_MATCH } from '../../operations/mutations';
-import { TextInputGroupBasic } from './TextInputBasic';
+import PropTypes from 'prop-types';
+import { ActionsColumn, Tr, Td } from '@patternfly/react-table';
+import { DELETE_MATCH, EDIT_MATCH } from '../../operations/mutations';
+import { formatDate, updateArray } from '../helpers';
 import { DropdownBasic } from './DropdownBasic';
+import { TextInputBasic } from './TextInputBasic';
+import { filterValues } from '../constants';
+import { useVerifyInput } from '../useVerifyInput';
 
-function formatDate(inputDate) {
-  const date = new Date(inputDate);
+export const NewMatchesTableRow = ({
+  columnNames,
+  data,
+  fetchDetailsTable,
+  fetchMatches,
+  hostId,
+  rowIndex,
+  setSelectedMatches,
+}) => {
+  const [isSelected, setIsSelected] = useState(false);
+  const [textValue, setTextValue] = useState(data.justification);
+  const [dropdownValue, setDropdownValue] = useState(data.status);
+  const [deleteMatch] = useMutation(DELETE_MATCH, {
+    variables: { input: [data.id] },
+  });
+  const [executeMutation] = useMutation(EDIT_MATCH);
 
-  const day = date.getUTCDate().toString().padStart(2, '0');
-  const month = date.toLocaleString('default', { month: 'long' });
-  const year = date.getUTCFullYear();
+  const onSelectMatch = (data) => {
+    setIsSelected(!isSelected);
+    setSelectedMatches((prev) => {
+      return updateArray(prev, data.id);
+    });
+  };
 
-  return `${day}-${month}-${year}`;
-}
+  const onDeleteMatch = async () => {
+    await deleteMatch();
+    await fetchDetailsTable(false);
+    await fetchMatches(hostId);
```

Comment:
I think here we have to setSelectedMatches to an empty array. This removes the issue where IDs persist in the array in between bulk selections